gensim/models/ldamodel.py

```diff
@@ -1054,13 +1059,22 @@ def load(cls, fname, *args, **kwargs):
             result.state = super(LdaModel, cls).load(state_fname, *args, **kwargs)
         except Exception as e:
             logging.warning("failed to load state from %s: %s", state_fname, e)
-        id2word_fname = utils.smart_extension(fname, '.id2word')
-        if (os.path.isfile(id2word_fname)):
-            try:
-                result.id2word = utils.unpickle(id2word_fname)
-            except Exception as e:
-                logging.warning("failed to load id2word dictionary from %s: %s", id2word_fname, e)
+
+        random_state_fname = utils.smart_extension(fname, '.random_state')
+        if (os.path.isfile(random_state_fname)):
+            result.random_state = utils.unpickle(random_state_fname)
         else:
-            result.id2word = None
+            logging.warning("random_state not stored on disk so using default value")
```

Comment:
@menshikh-iv Then there is no `.random_state` file involved at all, correct? To summarize, the solution is : 
- First, load the entire model.
- Check if `result.random_state` was set or not. For the newer (post 0.13.2) models, it would have been set through the main pickle load. For the older (pre 0.13.2) models, `result.random_state` would not be set through the main pickle load so we set `result.random_state` to `get_random_state(None)`.

And in this solution, we don't need to make any changes in the `save` function, just the `load` function.